website/static/js/pages/file-page.js

```diff
@@ -4,4 +4,44 @@ var $osf = require('js/osfHelpers');
 var FileViewPage = require('js/filepage');
 var waterbutler = require('js/waterbutler');
 
+require('jquery-tagsinput');
+
 m.mount(document.getElementsByClassName('file-view-panels')[0], FileViewPage(window.contextVars));
+
+function build_tag_url(nid, filePath, tag) {
+    return '/api/v1/project/' + nid + '/osfstorage' + filePath + '/tags/' + tag + '/';
+}
+
+$(function() {
```

Comment:
I don't know if that is possible/too useful, as the only part I would be able to reuse is up to ` /api/v1/project/nid `. If you think that that is something that should be reused, I can make that change. The ` build_tag_url ` would then look something like 
``` 
function build_tag_url(filePath, tag) {
    nodeApiUrl + '/osfstorage' + filePath + '/tags/' + tag + '/';
}
```
Not sure if that's what you had in mind.